src/txn/txn_rollback_to_stable.c

```diff
@@ -1728,9 +1740,8 @@ __rollback_to_stable_btree_apply(
      * 4. There is no durable timestamp in any checkpoint.
      * 5. The checkpoint newest txn is greater than snapshot min txn id.
      */
-    WT_WITH_HANDLE_LIST_READ_LOCK(session, (ret = __wt_conn_dhandle_find(session, uri, NULL)));
-
-    perform_rts = ret == 0 && S2BT(session)->modified;
+    WT_WITH_HANDLE_LIST_READ_LOCK(
```

Comment:
This change still holds the session with the dhandle pointer. You can change this also to call under WT_WITHOUT_DHANDLE, so that it replaces the existing dhandle after it finishes with that function. In this case, there is nothing to replace, so it sets to NULL.

Reply:
The underlying `WT_WITH_HANDLE_LIST_READ_LOCK` definition, grabs the connection handle list read lock. I don't think we can utilise `WT_WITHOUT_DHANDLE` here, as it just simply sets the dhandle back to NULL after the operation is finished, since we need to grab the lock.

Reply:
`WT_WITH_HANDLE_LIST_READ_LOCK` protects the dhandle until we check for whether the btree dirty or not. Without `WT_WITHOUT_DHANDLE` the session still points to a handle without any lock. This can lead to the same situation where the handle can be freed and we can access the freed memory while opening the history store.

Reply:
Just to clarify my comment, we need to use both macros. WT_WITHOUT_DHANDLE(WT_WITH_HANDLE_LIST_READ_LOCK(...));

Reply:
Should we have a new macro (i.e `WT_WITHOUT_HANDLE_LIST_READ_LOCK`)? Is that too confusing/counterintuitive?

Reply:
@kommiharibabu I have made this change. Along with this change I realised we don't need to call `WT_WITHOUT_DHANDLE` when we make the call to `__rollback_to_stable_btree_apply`, as the inner `WT_WITHOUT_DHANDLE` fixes the problem entirely.

> Should we have a new macro (i.e WT_WITHOUT_HANDLE_LIST_READ_LOCK)? Is that too confusing/counterintuitive?
I personally find that a little confusing etienne, I am happy make that change if you have a strong opinion about this!